src/usePerpsMetadata.ts

```diff
@@ -0,0 +1,27 @@
+import { useQuery } from '@tanstack/react-query';
+import { ethers } from 'ethers';
+import { useImportContract } from './useImports';
+import { useSynthetix } from './useSynthetix';
+
+interface PerpsMetadata {
+  name: string;
+  symbol: string;
+}
+
+export function usePerpsMetadata({ provider, marketId }: { provider?: ethers.providers.BaseProvider; marketId?: ethers.BigNumber }) {
```

Comment:
I believe we need to have more explicit distinction between perps/spot/core market ids by using more specific names
```suggestion
export function usePerpsMetadata({ provider, perpsMarketId }: { provider?: ethers.providers.BaseProvider; perpsMarketId?: ethers.BigNumber }) {
```

Reply:
Rename `marketId` to `perpsMarketId`.